### ISSUE TITLE
[TextInputLayout] Set Text Appearance Style For Hint in Expanded State

### ISSUE THREAD

**Issue by ShivanshPrakash** [material-components/material-components-android#2104]
**Is your feature request related to a problem? Please describe.**
A way to specify different attributes such as font size, letter spacing, font family, etc for hint text in expanded state (i.e when edit text is empty and unfocused). The current `hintTextAppearance` attribute seems to work only on floating/collapsed state.

**Additional context**
Will there be a problem if we allow `collapsingTextHelper` or other such properties of TextInputlayout class to be public so that they can be overridden in our own custom classes?

**Comment by leticiarossi** [comment]
If you set `android:textAppearance` on the edit text that will affect that expanded hint. It will also affect the text the user types, so if you want both to have different text appearances you'd have to do so programmatically - checking if the hint is expanded and calling `setTextAppearance` on the edit text accordingly. Does that help you?

**Comment by ShivanshPrakash** [comment]
Doing this programmatically works. But it would have been nice if it is directly supported by TextInputLayout itself.

**Comment by okarakose** [comment]
I agree with @ShivanshPrakash , I think hint's text appearance should only take care hintTextAppearance
